python/plugins/processing/core/parameters.py

```diff
@@ -79,13 +80,25 @@ def setValue(self, obj):
         of parameter.
         """
         if obj is None:
-            if not self.optional:
+            if not self.optional and self.default is None:
                 return False
-            self.value = None
-            return True
+            elif self.default is None:
+                self.value = None
+                return True
+            else:
```

Comment:
ah, ok, you mean initializating the default value. Yës, true, that can be done in __init__. If it's not like that, it is because originally not all parameters had the default value. I think now the default value is initialized, but not in the parent __init__, because of that.

Anyway, sound good to me, so if you guys want to do changes according to all this discussion, it will be fine. I think we have already enough info in this thread to make the implementation that is better and more robust :-)